src/Components/DateFacet.tsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import {
+  DateFacet as DateFacetType,
+  DateFacetState,
+  buildDateFacet,
+  DateFacetValue
+} from "@coveo/headless";
+import { headlessEngine } from "../Engine";
+import FormLabel from "@material-ui/core/FormLabel";
+import FormControl from "@material-ui/core/FormControl";
+import FormGroup from "@material-ui/core/FormGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import Box from "@material-ui/core/Box";
+
+interface IRangeDateFacetProps {
+  title: string;
```

Comment:
Not a super big deal, but since all types of facet components will have a required title and field, perhaps the IRangeDateFacetProps interface should extend a generic facet interface.